crates/volta-core/src/distro/node.rs

```diff
@@ -239,6 +250,17 @@ impl Distro for NodeDistro {
             dir: dest.clone(),
         })?;
 
+        if let Some(staging) = self.staging_file {
```

Comment:
In the refactor I suggested above, the invariant we maintain simply by way of checking it ourselves in the code flow here would be much more clearly required by the types themselves: if we had a `StagingDistro` and needed to transform it to a `NodeDistro` (or some such), this implementation would presumably live there, and the types themselves *wouldn't* let us treat a `StagingDistro` as a valid `NodeDistro` elsewhere.